### PR TITLE
Refresh directories and elements

### PR DESCRIPTION
**Pull request by bendaamerahm**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1623,8 +1623,6 @@
         },
```
Comment (sBouzols) [inline package-lock.json]: The package-lock.json shouldn't change in this PR I think.

File: src/components/tree-views-container.js [hunk 2]
```diff
@@ -495,16 +496,17 @@ const TreeViewsContainer = () => {
         return () => {
             wsToClose.close();
         };
-    }, []);
+    }, [dispatch]);
```
Comment (sBouzols) [inline src/components/tree-views-container.js]: there is no calls to dispatch in here... remove this dependency ? Then the effect will be run only on componentDidMount ?

File: src/utils/rest-api.js [hunk 3]
```diff
@@ -637,7 +637,7 @@ export function newScriptFromFiltersContingencyList(
  * Function will be called to connect with notification websocket to update the studies list
  * @returns {ReconnectingWebSocket}
  */
-export function connectNotificationsWsUpdateStudies() {
+export function connectNotificationsWsUpdateDirectory() {
```
Comment (sBouzols) [inline src/utils/rest-api.js]: a bit lower :
'Connected Websocket update ~~studies~~ directories '

File: src/redux/reducer.js [hunk 4]
```diff
@@ -134,4 +136,11 @@ export const reducer = createReducer(initialState, {
         delete newUploadingElements[action.uploadingElement.id];
         state.uploadingElements = newUploadingElements;
     },
+
+    [DIRECTORY_UPDATED]: (state, action) => {
+        state.directoryUpdated = {
+            force: 1 - state.directoryUpdated.force,
```
Comment (sBouzols) [inline src/redux/reducer.js:158]: Why this substraction ?

File: src/components/tree-views-container.js [hunk 5]
```diff
@@ -495,16 +496,17 @@ const TreeViewsContainer = () => {
...
 
     const onUpdateStudies = useCallback(
         (event) => {
             console.debug('Received Update Studies notification', event);
```
Comment (sBouzols) [inline src/components/tree-views-container.js]: `directories` then ?

File: src/components/tree-views-container.js [hunk 6]
```diff
@@ -495,16 +496,17 @@ const TreeViewsContainer = () => {
...
+
             let eventData = JSON.parse(event.data);
             if (!eventData.headers) {
                 return;
             }
-
+            dispatch(directoryUpdated(eventData));
```
Comment (sBouzols) [inline src/components/tree-views-container.js:512]: Maybe this is the only thing to do when initializing the websocket : 
```
wsRef.current.onmessage = (event) => { 
  let eventData = JSON.parse(event.data);
  if (!eventData.headers) {
      return;
  }
  dispatch(directoryUpdated(eventData));
};
```
Then the current onUpdateStudies function becomes an effect which proc when directoryUpdated changes in the store and consume differents types of notifications, you know like it's done in GridStudyApp

File: src/components/dialogs/directory-selector.js [hunk 7]
```diff
@@ -25,51 +28,153 @@ function sortAlphabetically(a, b) {
...
-        fetchDirectoryContent(nodeId).then((content) => {
-            addToDirectory(
-                nodeId,
-                content.filter((item) => contentFilter.has(item.type))
-            );
-            setData([...data]);
-        });
-    };
+    const fetchDirectory = useCallback(
+        (nodeId) => {
```
Comment (sBouzols) [inline src/components/dialogs/directory-selector.js]: Maybe rename nodeId to directoryUuid it's a bit confusing

File: src/components/dialogs/directory-selector.js [hunk 8]
```diff
@@ -25,51 +28,153 @@ function sortAlphabetically(a, b) {
...
+                    // update directory Content
+                    addToDirectory(
+                        nodeId,
+                        childrenToBeInserted.filter((item) =>
+                            contentFilter().has(item.type)
+                        )
+                    );
+                })
+                .catch((reason) => {
+                    console.warn(
```
Comment (sBouzols) [inline src/components/dialogs/directory-selector.js]: Maybe a snackbar output with a translated message instead of console warning ?

**Review by sBouzols — CHANGES_REQUESTED** [review]
Some code review comments, will test

File: src/redux/reducer.js (hunk 4 above)
Comment (bendaamerahm) [reply src/redux/reducer.js:158]: I did the same as gridStudy App

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK
Console Warning OK

**Review by YenguiSeddik — APPROVED** [review]
Test: ok
code review: ok

**Pull request merged**